javascript/features/death_match/death_match_manager.js

```diff
@@ -115,7 +148,78 @@ export class DeathMatchManger {
         }
 
         return spawnPositions[spawnIndex];
-    }    
+    }
+
+    onPlayerTakeDamage(event) {
+        const player = server.playerManager.getById(event.playerid);
+        if (!player)
+            return;  // the |event| was not received for a valid player
+
+        if (!this.playersInDeathMatch_.has(player.id))
+            return;
+
+        const issuer = server.playerManager.getById(event.issuerid);
+        if (!issuer)
+            return; // the |event| was not done by a valid player
+
+        if (!this.playersInDeathMatch_.has(issuer.id)) {
+            return;
+        }
+
+        if (isNaN(event.amount))
+            return; // the |event| does not have a valid amount of damage.
+
+        const issuerStats = this.playerStats_.get(issuer.id);
+        issuerStats.addDamage(+event.amount);
+    }
+
+    onPlayerWeaponShot(event) {
+        const player = server.playerManager.getById(event.playerid);
+        if (!player)
+            return;  // the |event| was not received for a valid |player|
+
+        if (!this.playersInDeathMatch_.has(player.id))
+            return;
+
+        const playerStats = this.playerStats_.get(player.id);
+        if (event.hittype === 1 || event.hittype === 2) {
+            playerStats.addBulletHit();
+        } else {
+            playerStats.addBulletMissed();
+        }
+    }
+
+    // If a player died we'll update stats and reward killer.
+    onPlayerDeath(event) {
+        const player = server.playerManager.getById(event.playerid);
+        if (!player)
+            return;  // the |player| couldn't be found, this is an invalid death
+
+        if (!this.playersInDeathMatch_.has(player.id))
+            return;
+
+        const killer = server.playerManager.getById(event.killerid);
+        if (!killer)
+            return;
+
+        if (!this.playersInDeathMatch_.has(killer.id)) {
+            return;
+        }
```

Comment:
consistency nit: either use {} for one-line statements or don't 😛

Reply:
I try to go according to the rest of th code (not having {}).
I'm just very used to using it thus forget sometimes :( Will remove :+1: